mas_perception_libs/common/include/mas_perception_libs/bounding_box_2d.h

```diff
@@ -0,0 +1,105 @@
+#include <utility>
+
+/*!
+ * @copyright 2018 Bonn-Rhein-Sieg University
+ *
+ * @author Minh Nguyen
+ *
+ */
+#ifndef MAS_PERCEPTION_LIBS_BOUNDING_BOX_2D_H
+#define MAS_PERCEPTION_LIBS_BOUNDING_BOX_2D_H
+
+#include <string>
+#include <vector>
+#include <opencv/cv.h>
+
+namespace mas_perception_libs
+{
+
+struct BoundingBox2D
+{
+    /*!
+     * @brief struct for representing a rectangle region in RGB images
+     */
+    int mX;
+    int mY;
+    int mWidth;
+    int mHeight;
+    std::string mLabel;
+    cv::Scalar mColor;
+
+    BoundingBox2D(const std::string &pLabel, const cv::Scalar &pColor, int pX, int pY, int pWidth, int pHeight)
+            : mX(pX), mY(pY), mWidth(pWidth), mHeight(pHeight), mLabel(pLabel), mColor(pColor)
+    { }
+
+    BoundingBox2D(const std::string &pLabel, int pX, int pY, int pWidth, int pHeight)
+            : BoundingBox2D(pLabel, CV_RGB(0, 0, 255), pX, pY, pWidth, pHeight)
+    { }
+
+    BoundingBox2D(int pX, int pY, int pWidth, int pHeight)
+            : BoundingBox2D("", pX, pY, pWidth, pHeight)
+    { }
+
+    BoundingBox2D() : BoundingBox2D(0, 0, 0, 0)
+    { }
+
+    /*! @brief update box geometry using a cv::Rect object */
+    void updateBox(const cv::Rect &);
+
+    /*! @brief get box geometry as a cv::Rect object */
+    cv::Rect getCvRect() { return cv::Rect(mX, mY, mWidth, mHeight); }
+};
+
+/*!
+ * @brief draw boxes on a CV image using BoundingBox2D objects
+ * @param pImage
+ * @param pBoundingBoxes: vector of bounding boxes to be drawn
+ * @param pThickness: line thickness
+ * @param pFontScale
+ */
+void
+drawLabeledBoxes(cv::Mat &pImage, std::vector<BoundingBox2D> pBoundingBoxes,
```

Comment:
the std::vector should be const reference here, right? Or is it  being modified?

Reply:
If the bounding boxes are modified, can the vectors be const ref?